### PR TITLE
Code that is `@AnnotatedFor("nullness")` is also `@AnnotatedFor("initialization")`

### PR DESCRIPTION
**Pull request by Ao-senXiong**

### CODE REVIEW

**Comment by Ao-senXiong** [comment]
Note: make the annotatedfor element follow alphabetical order

**Review by wmdietl — COMMENTED** [review]
Adding some more explanation to the PR description would make it easier to reconstruct the point of the PR in the future.

As discussed, alphabetically sorting the checkers in `AnnotatedFor` would be good. Is the IDE support that could do this? Could we have a simple Error Prone checker that looks for this and fixes it automatically?

**Comment by Ao-senXiong** [comment]
> Adding some more explanation to the PR description would make it easier to reconstruct the point of the PR in the future.
> 
> As discussed, alphabetically sorting the checkers in `AnnotatedFor` would be good. Is the IDE support that could do this? Could we have a simple Error Prone checker that looks for this and fixes it automatically?

@wmdietl Let me fix the CI failure first. Since this looks like a format issue, will an additional `spotless`  feature to format the elements in `annotatedfor` be more consistent to handle all format issue?

**Comment by Ao-senXiong** [comment]
@wmdietl  I am not sure why I can not assign you in this repo so I pin here. I hope this can go into release and we shall update CF in AWS.

**Comment by Ao-senXiong** [comment]
Don't merge at the moment as annotatedfor('initialization') did not work correctly. See https://github.com/eisop/checker-framework/issues/982

**Comment by wmdietl** [comment]
@Ao-senXiong Once https://github.com/eisop/checker-framework/pulls/985 and https://github.com/eisop/checker-framework/pull/991 are merged, can this be merged?

File: src/java.base/share/classes/java/io/BufferedWriter.java [hunk 1]
```diff
@@ -70,7 +70,7 @@
  * @since       1.1
  */
 
-@AnnotatedFor({"index", "mustcall", "nullness"})
+@AnnotatedFor({"index", "mustcall", "nullness", "initialization"})
```
Comment (wmdietl) [inline src/java.base/share/classes/java/io/BufferedWriter.java:73]:
```suggestion
@AnnotatedFor({"index", "initialization", "mustcall", "nullness"})
```

File: src/java.base/share/classes/java/io/BufferedInputStream.java [hunk 2]
```diff
@@ -58,7 +58,7 @@
  * @author  Arthur van Hoff
  * @since   1.0
  */
-@AnnotatedFor({"index", "nullness", "mustcall"})
+@AnnotatedFor({"index", "initialization", "nullness", "mustcall"})
```
Comment (wmdietl) [inline src/java.base/share/classes/java/io/BufferedInputStream.java:61]:
```suggestion
@AnnotatedFor({"index", "initialization", "mustcall", "nullness"})
```

File: src/java.base/share/classes/java/io/BufferedOutputStream.java [hunk 3]
```diff
@@ -42,7 +42,7 @@
  * @author  Arthur van Hoff
  * @since   1.0
  */
-@AnnotatedFor({"index", "nullness", "mustcall", "signedness"})
+@AnnotatedFor({"index", "initialization", "nullness", "mustcall", "signedness"})
```
Comment (wmdietl) [inline src/java.base/share/classes/java/io/BufferedOutputStream.java:45]:
```suggestion
@AnnotatedFor({"index", "initialization", "mustcall", "nullness", "signedness"})
```

File: src/java.base/share/classes/java/io/CharArrayWriter.java [hunk 4]
```diff
@@ -49,7 +49,7 @@
  * @author      Herb Jellinek
  * @since       1.1
  */
-@AnnotatedFor({"index", "lock", "mustcall", "nullness"})
+@AnnotatedFor({"index", "lock", "mustcall", "nullness", "initialization"})
```
Comment (wmdietl) [inline src/java.base/share/classes/java/io/CharArrayWriter.java:52]:
```suggestion
@AnnotatedFor({"index", "initialization", "lock", "mustcall", "nullness"})
```

File: src/java.base/share/classes/java/io/File.java [hunk 5]
```diff
@@ -42,7 +42,9 @@
 import java.net.URISyntaxException;
 import java.nio.file.FileStore;
 import java.nio.file.FileSystems;
+import java.nio.file.LinkOption;
```
Comment (wmdietl) [inline src/java.base/share/classes/java/io/File.java:45]: Why are the additional imports needed?

File: src/java.base/share/classes/java/io/FileInputStream.java [hunk 6]
```diff
@@ -69,7 +69,7 @@
  * @see     java.nio.file.Files#newInputStream
  * @since   1.0
  */
-@AnnotatedFor({"index", "mustcall", "nullness"})
+@AnnotatedFor({"index", "initialization", "mustcall", "nullness",})
```
Comment (wmdietl) [inline src/java.base/share/classes/java/io/FileInputStream.java:72]:
```suggestion
@AnnotatedFor({"index", "initialization", "mustcall", "nullness"})
```

File: src/java.base/share/classes/java/util/EnumSet.java [hunk 7]
```diff
@@ -79,7 +79,7 @@
  * @since 1.5
  * @see EnumMap
  */
-@AnnotatedFor({"nullness", "index"})
+@AnnotatedFor({"initialization", "nullness", "index"})
```
Comment (wmdietl) [inline src/java.base/share/classes/java/util/EnumSet.java:82]:
```suggestion
@AnnotatedFor({"index", "initialization", "nullness"})
```

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks!

File: src/java.base/share/classes/java/io/File.java (hunk 5 above)
Comment (Ao-senXiong) [reply src/java.base/share/classes/java/io/File.java:45]: Ah, I think ide must do this automatically... I never did that myself. I see a few places in the comment. I will remove them.

**Review by wmdietl — APPROVED** [review]

**Comment by wmdietl** [comment]
@Ao-senXiong I just saw [this file](https://github.com/eisop/jdk/blob/dc094144ce01dca6ec3c9cc6659f0b55b0dd192b/src/java.base/share/classes/java/util/Optional.java#L88) that is only `AnnotatedFor` for nullness, but not for initialization.

Is there a reason for this?
Can you write a script that makes it easier to validate questions about `AnnotatedFor`? E.g. as part of the [JavaStubifier](https://github.com/eisop/checker-framework/blob/master/framework/src/stubifier/java/org/checkerframework/framework/stubifier/JavaStubifier.java) we should relatively easily be able to add such a consistency check. We probably don't want to always enforce this property, but should at least add a demo for how to do it.

Also, please add a comment somewhere that when somebody adds nullness annotations, they should also think about adding initialization.

**Comment by Ao-senXiong** [comment]
I have Created #120 and https://github.com/eisop/checker-framework/issues/1151 for above tasks.

**Pull request merged**